app/main.py

```diff
@@ -1,34 +1,81 @@
+from math import dist
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: tuple, end: tuple,
+                 is_drowned: bool = False) -> None:
+        self.start = start
+        self.end = end
+        self._distance = int(dist(end, self.start))
+        self.decks = self._deck_builder()
+        self.is_drowned = is_drowned
+
+    def _deck_builder(self) -> list:
+        if self.start == self.end:
+            decks = [Deck(self.start[0], self.start[1])]
+        elif self.start[0] == self.end[0]:
+            decks = [Deck(
+                self.start[0], self.start[1] + number
+            ) for number in range(self._distance + 1)]
+        else:
+            decks = [Deck(
+                self.start[0] + number, self.start[1]
+            ) for number in range(self._distance + 1)]
+        return decks
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(self, row: int, column: int) -> bool:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck.is_alive
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(self, row: int, column: int) -> None:
+        counter = 0
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                deck.is_alive = False
+            if not deck.is_alive:
+                counter += 1
+
+        if len(self.decks) == counter:
+            self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list) -> None:
```

Comment:
add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contain string elements has typehint list[str]